src/software/ai/hl/stp/play/shoot_or_pass/shoot_or_pass_play_fsm.cpp

```diff
@@ -54,12 +53,13 @@ void ShootOrPassPlayFSM::lookForPass(const Update& event)
         auto pitch_division =
             std::make_shared<const EighteenZonePitchDivision>(event.common.world.field());
 
-        auto pass_eval    = pass_generator.generatePassEvaluation(event.common.world);
-        auto ranked_zones = pass_eval.rankZonesForReceiving(
-            event.common.world, event.common.world.ball().position());
+        auto pass_eval = pass_generator.generatePassEvaluation(event.common.world);
 
-        best_pass_and_score_so_far = pass_eval.getBestPassOnField();
+        auto best_pass_score_and_zone = pass_eval.getBestPassAndZoneOnField();
+        best_pass_zone                = best_pass_score_and_zone.first;
+        best_pass_and_score_so_far    = best_pass_score_and_zone.second;
 
+        auto ranked_zones = getBestOffensivePositions(pass_eval, event);
```

Comment:
to confirm, are these the zones for receivers?

Reply:
yes these are the best zones to receive except the one when the receiver robot is

so all supporting zones